### PR TITLE
React tutorial: Todo app

### PR DESCRIPTION
**Pull request by aliabbani**
## React tutorial: Todo app

In this milestone we:

- Create class components and make the web app work.
- Add different styling methods.
- Refactored the component from a class-based component to a functional component with hooks
- After the refactor is done, the app keeps the same functionality.
- Add React router and create a SPA.

### CODE REVIEW

File: src/functionBased/components/TodoItem.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  // useEffect(() => () => {
+  //   console.log('cleaning up...');
+  // }, []);
+
```
Comment (safafa) [inline src/functionBased/components/TodoItem.js]: - [x] Kindly remove the commented code to increase the code readability.

**Review by safafa — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED ♻️ 

Hi @aliabbani  You did a great job so far 👏🏼 👏🏼 

- The app is working as expected.
- The linter checks are passing.
- the pull request is descriptive.

However, here are some points that you need to work on to get your project approved.

## Required changes :

- [x] Check the inline comments.
- [x] Kindly update the Readme file. To make it more descriptive you can add a short description of the project, the project screenshot. You can also remove the extra react auto-generated text and only keep the useful ones.

## Optional changes :
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding more style to the application

Cheers and Happy coding!👏👏👏

File: src/functionBased/components/TodoItem.js (hunk 1 above)
Comment (aliabbani) [reply src/functionBased/components/TodoItem.js]: Hello, @safafa thank you for your valuable comment. I already deleted the commented code.

Comment (aliabbani) [reply src/functionBased/components/TodoItem.js]: I also update my readme file upon your request 😄

**Review by safafa — APPROVED** [review]
## STATUS: APPROVED 🟢 👏
Hello, @aliabbani  great job working on the required changes 👍 💯

Now you can merge this branch 💪 😄
HAPPY CODING!! AND GOOD LUCK WITH THE NEXT ONES 🤞 🍀

**Pull request merged**